### ISSUE TITLE
Portuguese support

### ISSUE THREAD

**Issue by arademaker** [flairNLP/flair#2112]
To be removed, once it is done: Please add the appropriate label to this ticket, e.g. feature or enhancement.

**Is your feature/enhancement request related to a problem? Please describe.**

I would like to use for Portuguese data. Documentation is not clear about how to define the language when processing documents. 

**Describe the solution you'd like**

1. better documentation about language support
2. how to setup the pipeline for a specific language
3. in/out formats

**Comment by alanakbik** [comment]
Hi @arademaker what would you like to do on Portuguese? Train / use a tagger (like NER or POS), or train your own embeddings?

**Comment by arademaker** [comment]
I have been working on a collection of texts from a bibliographic database called DHBB. The text files are in http://github.com/cpdoc/dhbb-nlp/. We are exploring the construction of a corpus from these files. Recent paper at http://arademaker.github.io/bibliography/propor-2020.html

The idea is to add as many layers of annotation as possible. The first step is:

1. split the texts into sentences
2. tokenization, lemmatization, POS, dependency parser
3. revise (2) cross-checking with NER from IBM Watson Knowledge Studio and https://github.com/LFG-PTBR/MorphoBr 

The paper above describes how we found many problems in sentence split combining more than one tool. For tokenization, lemmatization, POS, and dependencies, we have been using UDPipe that is easy to train (in cycles, once we find errors we fix them and add the manual annotations to the training set). But UDPipe has some limitations, so we are starting to investigate alternatives like Stanza and Flair.

Once we have good results above,  we may add more layers and move the conllu to conllup files:

4. enhanced deps
5. SRL
6. MWE expressions
7. word sense disambiguation with http://openwordnet-pt.org

**Comment by alanakbik** [comment]
There are currently no pre-trained models for Portuguese shipped with Flair (except one for clinical POS tagging), but its very easy to train your own models. 

For instance, to train a POS tagger for Portuguese, you can use the following script: 

```python
from flair.datasets import UD_PORTUGUESE
from flair.embeddings import StackedEmbeddings, FlairEmbeddings

# 1. load the Prortuguese UD treebank
corpus = UD_PORTUGUESE()

# 2. what tag do we want to predict?
tag_type = 'pos' #could also be 'upos' for universal POS or even mophological tags like 'number'

# 3. make the tag dictionary from the corpus
tag_dictionary = corpus.make_tag_dictionary(tag_type=tag_type)
print(tag_dictionary)

# 4. initialize each embedding we use
embedding_types = [

    # contextual string embeddings, forward
    FlairEmbeddings('pt-forward'),

    # contextual string embeddings, backward
    FlairEmbeddings('pt-backward'),
]

# embedding stack consists of Flair forward and backward embeddings
embeddings = StackedEmbeddings(embeddings=embedding_types)

# 5. initialize sequence tagger
from flair.models import SequenceTagger

tagger = SequenceTagger(hidden_size=256,
                        embeddings=embeddings,
                        tag_dictionary=tag_dictionary,
                        tag_type=tag_type,
                        use_crf=False) # maybe set this to True

# 6. initialize trainer
from flair.trainers import ModelTrainer

trainer = ModelTrainer(tagger, corpus)

# 7. run training
trainer.train('resources/taggers/pos-portuguese',
              train_with_dev=True,
              max_epochs=150)
```

If you train good models, we'd be happy if you shared them with the community! There is a new feature coming in collaboration with HuggingFace where you can upload your Flair models to the [HF model hub](https://huggingface.co/models?filter=flair).

If you want to better understand how to train models, I'd recommend going through the [Flair tutorials](https://github.com/flairNLP/flair#tutorials), in particular tutorials 1-3, and 6-7 in your case!
